### PR TITLE
Some enhancements!

### PR DESCRIPTION
**Pull request by devilx**
- Implementation of DISTINCT option
- Implementation of protocolOptions (needed it to change the port)
- Some debug output, as I had some trouble and had to start digging quite deep to find out what was wrong -> my schema
- Implemented matching through the decomposed array and noticed you just did the same, so I picked yours now. :)
- Implemented possibility to have uppercase KEYSPACE names
- Extended timeuuid and implemented maxTimeuuid & minTimeuuid methods (tests for those are missing so far, will do that asap)

### CODE REVIEW

**Comment by devilx** [comment]
Not sure why the build fails here:

```

  1) Unit Tests #modelsync should connect and sync with db without errors:
     Error: timeout of 5000ms exceeded. Ensure the done() callback is being called in this test.

```

**Comment by masumsoft** [comment]
@devilx Thanks for the PR. For some reason the tests timed out in Travis CI's servers. Did run the tests again and it's passing now. 

I'm a bit busy at this moment, will try to have a more thorough look at the PR tomorrow.

File: README.md [hunk 1]
```diff
@@ -161,11 +167,11 @@ john.delete(function(err){
...
 Express cassandra exposes some node driver methods for convenience. To generate uuids e.g. in field defaults:
 
-*   `models.uuid()`  
+*   `models.uuid()`
     returns a type 3 (random) uuid, suitable for Cassandra `uuid` fields, as a string
-*   `models.uuidFromString(str)`  
+*   `models.uuidFromString(str)`
     returns a type 3 uuid from input string, suitable for Cassandra `uuid` fields
-*   `models.timeuuid()`  
+*   `models.timeuuid()`
```
Comment (masumsoft) [inline README.md]: Add documentation for maxTimeuuid and minTimeuuid

File: lib/orm/apollo_schemer.js [hunk 2]
```diff
@@ -157,8 +161,10 @@ var schemer = {
         else if(typeof fieldob == 'object'){
             return fieldob.type;
         }
-        else
+        else {
+            throw("Uknown field '" + fieldname + "'");
             return undefined;
```
Comment (masumsoft) [inline lib/orm/apollo_schemer.js]: Shouldn't this throw a message something like the following?

```
throw("Field type not defined for '" + fieldname + "'");
```

Could you please explain the scenario where an `Unknown field` error message would be appropriate?

File: lib/orm/base_model.js [hunk 3]
```diff
@@ -28,7 +28,37 @@ var BaseModel = function(instance_values){
...
+                } else if(typeof fields[prop_name].toLowerCase != "undefined") {
+                    fieldDataType = fields[prop_name].toLowerCase();
+                }
+                switch(fieldDataType) {
+                    case "int":
+                    case "bigint":
+                    case "varint":
+                    case "double":
+                    case "counter":
+                        retval = parseInt(this[prop_name]);
```
Comment (masumsoft) [inline lib/orm/base_model.js]: I didn't get why bigint, double and counter types are parsed as integers.

File: lib/orm/base_model.js [hunk 4]
```diff
@@ -742,7 +772,8 @@ BaseModel._create_find_query = function(query_ob, options){
     }
 
     var query = util.format(
-        'SELECT %s FROM "%s" %s %s %s',
+        'SELECT %s %s FROM "%s" %s %s %s',
+        (options.distinct ? "DISTINCT" : ""),
```
Comment (masumsoft) [inline lib/orm/base_model.js]: Add the distinct option in Readme docs.

**Comment by masumsoft** [comment]
@devilx did some review of the PR, could you please address the issues in the diff comments?

File: lib/orm/base_model.js (hunk 3 above)
Comment (devilx) [inline lib/orm/base_model.js]: You're right, Sir. Totally forgot about JavaScript's limitations. And of course, double must have ended at the parseFloat block, if any. Will rewrite this.

File: lib/orm/apollo_schemer.js (hunk 2 above)
Comment (devilx) [inline lib/orm/apollo_schemer.js]: :+1:

**Comment by devilx** [comment]
@masumsoft so, I basically removed the "more intelligent" stuff from the framework code as it really wasn't that intelligent after all. I now built it into _my_ application using two extensions: One that "stringiefies" the Model for output purpose, so that JSON is always generated with numbers as JSON-strings; the other extension reads the JSON data (expecting numbers as strings as well) and converts them according to the fields-structure, by using `require('cassandra-driver').types.*` ([see here](https://github.com/datastax/nodejs-driver/tree/master/lib/types)).

However, I don't think it makes sense to force people using express-cassandra to follow this behaviour (only sending and expecting strings). Therefor I reverted the changes I did to the getters.

The requested documentation was also added, btw.

Regards!

**Comment by masumsoft** [comment]
Merged. Could you please add some test cases for the DISTINCT, maxTimeuuid & minTimeuuid options before we can ship this for the next release?

**Pull request merged**